hastory-server-data/src/Hastory/Server/Data/Username.hs

```diff
@@ -1,48 +1,32 @@
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
 {-# LANGUAGE OverloadedStrings #-}
 
 module Hastory.Server.Data.Username where
 
 import Data.Aeson (FromJSON(parseJSON), ToJSON(toJSON), withText)
-import qualified Data.CaseInsensitive as CI
 import Data.Char
-import Data.Proxy (Proxy(Proxy))
+import Data.Text (Text)
 import qualified Data.Text as T
 import Data.Validity
-import Database.Persist (PersistField(fromPersistValue, toPersistValue), PersistValue(PersistText))
-import Database.Persist.Sql (PersistFieldSql(sqlType))
+import Database.Persist.Sql (PersistField, PersistFieldSql)
 
 newtype Username =
   Username
-    { unUsername :: CI.CI T.Text
+    { usernameText :: Text
     }
   deriving (Eq)
-
-instance PersistField Username where
-  toPersistValue = toPersistValue . rawUserName
-  fromPersistValue (PersistText userName) = Right $ mkUsername userName
-  fromPersistValue _ = Left "Username value must be converted from PersistText"
-
-instance PersistFieldSql Username where
-  sqlType _ = sqlType (Proxy :: Proxy T.Text)
-
-instance Show Username where
-  show = show . rawUserName
+  deriving newtype (Show, PersistField, PersistFieldSql)
 
 instance FromJSON Username where
-  parseJSON = withText "Username" $ pure . mkUsername
+  parseJSON = withText "Username" $ pure . Username
 
 instance ToJSON Username where
-  toJSON = toJSON . rawUserName
+  toJSON = toJSON . usernameText
 
 instance Validity Username where
   validate userName = mconcat [check notNull "Username cannot be null", allAlphaNum]
     where
-      notNull = not . T.null . rawUserName $ userName
+      notNull = not . T.null . usernameText $ userName
       allAlphaNum =
-        decorateList (T.unpack . rawUserName $ userName) (declare "is alpha-numeric" . isAlphaNum)
-
-mkUsername :: T.Text -> Username
-mkUsername = Username . CI.mk
```

Comment:
I think we still need a `parseUsername :: Text -> Maybe Username` but it has to be able to fail. You can use `constructValid`: https://hackage.haskell.org/package/validity-0.11.0.0/docs/Data-Validity.html#v:constructValid

Reply:
I've deleted `mkUsername` in a previous PR given that we removed the entire concept of making the username case insensitive.